app/models/pet.rb

```diff
@@ -10,4 +12,8 @@ def shelter_name
   def self.adoptable
     where(adoptable: true)
   end
+
+  def self.search_for_pet(search)
+    where('lower(name) like ?', "%#{search.downcase}%")
```

Comment:
This search method is actually already implemented for you in the ApplicationRecord! This is a bit of an Easter Egg (although you're not penalized for not finding it!). Because this is a brownfield project, we want you to explore the codebase and leverage the methods already created (like you did with  the `error_messages` helper method in ApplicationController). So, you could've used this `.search` method from the model parent class and just tested it in order to complete this user story

Reply:
Actually, it looks like your tests are testing the `.search` function that's already created instead of this method, so you can remove this method and make sure you're calling .search from your controller!